### ISSUE TITLE
Running debugger always opens 'xterm' regardless of terminal setting

### ISSUE THREAD

**Issue by jcowgill** [eranif/codelite#622]
From https://bugs.debian.org/773396

When running a program in the debugger, codelite always opens an instance of ```xterm``` to handle the console input regardless of the setting in Global Editor Preferences -> Terminal. I think debugging should use the terminal setting as well (like the other ways you can open a terminal from within codelite).

**Comment by eranif** [comment]
`xterm` has some unique features that we count on (we execute a `sleep` command and then capturing the `tty`). So we rely on the fact that there is a switch that can be used to execute commands directly in the shell ( I am sure other console support it as well, but we don't know the type of the console the user chose)

We could run some code to try to do the same for other consoles, but I am not sure it can be done easily.

Long story short: its not worth my (or David's effort). I urge you (or the OP of the Debian bug) to submit a PR)


Thanks
Eran
